internal/cephfs/clone.go

```diff
@@ -94,11 +94,11 @@ func createCloneFromSubvolume(ctx context.Context, volID, cloneID volumeID, volO
 	}
 
 	cloneState, cloneErr := volOpt.getCloneState(ctx, cloneID)
-	if cloneErr != nil {
```

Comment:
@yati1998 looks like we don't need to remove this one. there could be a corner case `cloneErr` might be nil and `cloneState` is empty return new error type when the case is `case cephFSCloneError`

Reply:
@nixpanic what do you think?

Reply:
sure @Madhu-1

Reply:
I haven't tried, but can this be expressed like:
```go
case cephFSCloneError || cloneErr != nil:
```

Reply:
If it is confirmed that `cloneState` is set to `cephFSCloneError` by default in `getCloneState()`, then checking the returned error is not strictly required (but still safer).

Reply:
again we need to check `cloneErr` is nil inside the case to decide or error message? or suggestion is to append error message to some const like `invalid %s  clone status with cloneErr %w` something  like this?

Reply:
Yes, this check is to be on the safer side to avoid unknown issues.

Reply:
> again we need to check `cloneErr` is nil inside the case to decide or error message? or suggestion is to append error message to some const like `invalid %s clone status with cloneErr %w` something like this?

@Madhu-1 We can't check cloneErr inside the case as it is of a different type. Hence adding the check as it was earlier with a log message.